CRM/Utils/Date.php

```diff
@@ -2136,7 +2163,7 @@ public static function formatDate($date, $dateType) {
       if ($valid) {
         // format date and time to default.
         if ($ruleName == 'dateTime') {
-          $dateVal = CRM_Utils_Date::customFormat(preg_replace("/(:|\s)?/", "", $dateVal), '%Y%m%d%H%i');
+          $dateVal = CRM_Utils_Date::customFormat(preg_replace("/(:|\s)?/", "", $dateVal), '%Y%m%d%H%i%S');
           // hack to add seconds
           $dateVal .= '00';
         }
```

Comment:
Thanks for the patch. Just noticing right below this line it adds '00' for seconds.

Also I think line 2152 above has come up before as doing something weird where it's comparing against a time format without seconds. I'll see if I can find that but it might be relevant here too.